rcl/src/rcl/rmw_implementation_identifier_check.c

```diff
@@ -146,12 +153,20 @@ rcl_ret_t rcl_rmw_implementation_identifier_check(void)
         actual_rmw_impl_id,
         RCL_RET_MISMATCHED_RMW_ID
       );
-      return RCL_RET_MISMATCHED_RMW_ID;
+      ret = RCL_RET_MISMATCHED_RMW_ID;
+      goto cleanup;
     }
-    // Free the memory now that all checking has passed.
+  }
+  ret = RCL_RET_OK;
+// fallthrough
+cleanup:
+  if (expected_rmw_impl) {
     allocator.deallocate((char *)expected_rmw_impl, allocator.state);
   }
```

Comment:
nit:
```suggestion
  allocator.deallocate(expected_rmw_impl, allocator.state);
```

- `allocator.deallocate()` is a noop when passing `NULL`, so no need of the if (I forgot that in my previous suggestion :smile: )
- `allocator.deallocate()` takes a `void *`, casting to `char *` doesn't make sense.

Reply:
Sure, see 7f0ffeb